pyanaconda/dbus/__init__.py

```diff
@@ -47,25 +48,30 @@ def get_connection():
         return DBus._connection
 
     @staticmethod
-    def get_new_connection():
+    def get_new_connection(bus_address=None):
         """Get a DBus connection.
 
         You shouldn't create new connections unless there is a good
         reason for it. Use DBus.get_connection instead.
 
-        Normally this method should return a connection to the system
+        Normally this method should return a connection to the session
         bus, but during testing/development a custom bus might be used.
         So just always connect to the bus specified by the environmental
         variable DBUS_STARTER_ADDRESS.
-        """
-        bus_address = os.environ.get("DBUS_STARTER_ADDRESS")
-
-        if bus_address:
-            log.info("Connecting to DBus at %s.", bus_address)
-            return pydbus.connect(bus_address)
 
-        log.info("Connecting to system DBus.")
-        return pydbus.SystemBus()
+        :param bus_address: Define your specific DBus session.
+                            This parameter is useful especially for testing.
+        """
+        if not bus_address:
+            if DBUS_STARTER_ADDRESS in os.environ:
+                bus_address = os.environ.get(DBUS_STARTER_ADDRESS)
+            elif DBUS_SESSION_ADDRESS in os.environ:
```

Comment:
Have you consider to set up the `DBUS_STARTER_ADDRESS` for the main anaconda process, so we don't have to handle here `DBUS_SESSION_ADDRESS`?

Reply:
But it is a lie, It is not started by DBUS. I think it is not that hard to set `DBUS_SESSION_ADDRESS` instead.